tests/model/SiteTreeTest.php

```diff
@@ -108,6 +109,47 @@ public function testURLGeneration()
     }
 
     /**
+     * Check if reserved URL's are properly appended with a number at top level
+     */
+    public function testDisallowedURLGeneration()
+    {
+        // Expected URL defaults to '-2'
+        $expectedURLs = [
```

Comment:
I wouldn't mind it if you just added these for these new tests and ignored the older ones. ;)